kolibri/core/public/api.py

```diff
@@ -146,3 +156,77 @@ def get_public_file_checksums(request, version):
         json.dumps({"id": error_constants.NOT_FOUND, "metadata": {"view": ""}}),
         content_type="application/json",
     )
+
+
+class SyncQueueViewSet(viewsets.ViewSet):
+    def list(self, request):
+        """Returns length of the queue for each of the available facilities"""
+        SyncQueue.clean_stale()  # first, ensure not expired devices are in the queue
+        facilities = Facility.objects.all()
+        queue = {}
+        for facility in facilities:
+            queue[facility.id] = SyncQueue.objects.filter(
+                user__facility=facility
+            ).count()
+        return Response(queue)
+
+    def check_queue(self):
+        current_transfers = TransferSession.objects.filter(active=True).count()
+        if current_transfers <= MAX_CONCURRENT_SYNCS:
+            allow_sync = True
+            data = {"action": SYNC}
+        else:
+            polling = current_transfers + SyncQueue.objects.all().count()
```

Comment:
Might be good to add a constant here for the 'waiting time' and use this as a multiplier, or even be calculating some sort of syncing estimate (we'd need to check in with @bjester and @jamalex to see if that would be feasible).